packages/amplify_auth_cognito/example/ios/amplify_auth_cognito_tests/amplify_auth_cognito_tests.swift

```diff
@@ -0,0 +1,1168 @@
+                let codeDeliveryJson = ((res.toJSON()["nextStep"] as! [String: Any])["codeDeliveryDetails"] as! [String: String])
+                XCTAssertEqual(_phoneNumber, codeDeliveryJson["destination"]!)
+
+            } else {
+                XCTFail()
+            }
+        })
+    }
+    
+    func test_signUpSuccessCustom() {
+        
+        class SignUpMock: AuthCognitoBridge {
+            override func onSignUp(flutterResult: @escaping FlutterResult, request: FlutterSignUpRequest){
+                let signUpRes = Result<AuthSignUpResult,AuthError>.success(
+                    AuthSignUpResult(AuthSignUpStep.confirmUser(AuthCodeDeliveryDetails(destination: DeliveryDestination.unknown(_phoneNumber)), ["foo": "bar"])))
+                let signUpData = FlutterSignUpResult(res: signUpRes)
+                flutterResult(signUpData)
+            }
+        }
+        
+        plugin = SwiftAuthCognito.init(cognito: SignUpMock())
+        
+        _attributes = ["custom" : _phoneNumber]
+        _options = ["userAttributes": _attributes]
+        _data = [
+            "username": _username,
+            "password": _password,
+            "options": _options
+        ]
+        _args = ["data": _data]
+        let call = FlutterMethodCall(methodName: "signUp", arguments: _args)
+        plugin.handle(call, result: {(result)->Void in
+            if let res = result as? FlutterSignUpResult {
+                XCTAssertEqual( false, res.isSignUpComplete )
+                XCTAssertEqual( "CONFIRM_SIGN_UP_STEP", res.signUpStep)
+                let codeDeliveryJson = ((res.toJSON()["nextStep"] as! [String: Any])["codeDeliveryDetails"] as! [String: String])
+                XCTAssertEqual(_phoneNumber, codeDeliveryJson["destination"]!)
+
+            } else {
+                XCTFail()
+            }
+        })
+    }
+    
+    func test_signUpSuccessComplete() {
+        
+        class SignUpMock: AuthCognitoBridge {
+            override func onSignUp(flutterResult: @escaping FlutterResult, request: FlutterSignUpRequest){
+                let signUpRes = Result<AuthSignUpResult,AuthError>.success(AuthSignUpResult(AuthSignUpStep.done))
+                let signUpData = FlutterSignUpResult(res: signUpRes)
+                flutterResult(signUpData)
+            }
+        }
+        
+        plugin = SwiftAuthCognito.init(cognito: SignUpMock())
+        
+        _attributes = ["email" : _email]
+        _options = ["userAttributes": _attributes, "complete": true]
+        _data = [
+            "username": _username,
+            "password": _password,
+            "options": _options
+        ]
+        _args = ["data": _data]
+        let call = FlutterMethodCall(methodName: "signUp", arguments: _args)
+        plugin.handle(call, result: {(result)->Void in
+            if let res = result as? FlutterSignUpResult {
+                XCTAssertEqual( "DONE", res.signUpStep)
+            } else {
```

Comment:
Should we add a `XCTAssertEqual(true/false, res.isSignUpComplete)` here?

Reply:
Done.